### PR TITLE
env enhancements

### PR DESCRIPTION
**Pull request by evgeniiz321**

### CODE REVIEW

File: neofs-testlib/neofs_testlib/env/templates/neofs_env_config.yaml [hunk 1]
```diff
@@ -1,49 +1,44 @@
...
         file: 'neofs-rest-gw-linux-amd64'
 
-    neofs_http_gw:
-        repo: 'nspcc-dev/neofs-http-gw'
-        version: 'v0.28.0'
-        file: 'neofs-http-gw-linux-amd64'
-
     neo_go:
         repo: 'nspcc-dev/neo-go'
         version: 'v0.104.0'
```
Comment (smallhive) [inline neofs-testlib/neofs_testlib/env/templates/neofs_env_config.yaml:44]: I think we can use newer version `v0.106.0`

**Review by smallhive — APPROVED** [review]

File: neofs-testlib/neofs_testlib/env/templates/neofs_env_config.yaml [hunk 2]
```diff
@@ -1,49 +1,44 @@
 binaries:
     neofs_cli:
         repo: 'nspcc-dev/neofs-node'
-        version: 'v0.39.2'
+        version: 'v0.42.0'
         file: 'neofs-cli-amd64'
```
Comment (smallhive) [inline neofs-testlib/neofs_testlib/env/templates/neofs_env_config.yaml:5]: In the latest releases the file name has been changed and requires to be updated as well:
* `neofs-cli-amd64` -> `neofs-cli-linux-amd64`
* `neofs-adm-amd64` -> `neofs-adm-linux-amd64`
* `neofs-ir-amd64` -> `neofs-ir-linux-amd64`
* `neofs-lens-amd64` -> `neofs-lens-linux-amd64`
* `neofs-node-amd64` -> `neofs-node-linux-amd64`

File: neofs-testlib/neofs_testlib/env/env.py [hunk 3]
```diff
@@ -42,6 +43,7 @@ class WalletType(Enum):
 
 
 class NeoFSEnv:
+    _id = f"{datetime.datetime.now()}".replace(" ", "-").replace(":", "-").replace(".", "-")
```
Comment (roman-khimov) [inline neofs-testlib/neofs_testlib/env/env.py:46]: Doesn't look reliable, format your date and time manually if you want to be sure in anything.

File: neofs-testlib/neofs_testlib/env/templates/neofs_env_config.yaml (hunk 1 above)
Comment (evgeniiz321) [reply neofs-testlib/neofs_testlib/env/templates/neofs_env_config.yaml]: fixed

File: neofs-testlib/neofs_testlib/env/templates/neofs_env_config.yaml (hunk 2 above)
Comment (evgeniiz321) [reply neofs-testlib/neofs_testlib/env/templates/neofs_env_config.yaml]: fixed

File: neofs-testlib/neofs_testlib/env/env.py (hunk 3 above)
Comment (evgeniiz321) [reply neofs-testlib/neofs_testlib/env/env.py]: fixed

File: neofs-testlib/neofs_testlib/env/env.py [hunk 4]
```diff
@@ -713,13 +722,8 @@ def _launch_process(self):
         stderr_fp = open(self.stderr, "w")
         s3_gw_env = {
             "S3_GW_LISTEN_DOMAINS": self.neofs_env.domain,
```
Comment (smallhive) [inline neofs-testlib/neofs_testlib/env/env.py:726]: Missed it, could we completely remove ENV vars for the gate? I mean move this setting also to yaml config file

**Review by smallhive — APPROVED** [review]

File: neofs-testlib/neofs_testlib/env/env.py (hunk 4 above)
Comment (evgeniiz321) [reply neofs-testlib/neofs_testlib/env/env.py]: fixed

**Review by smallhive — APPROVED** [review]

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**